ogr/utils.py

```diff
@@ -4,24 +4,111 @@
 import functools
 import logging
 import re
+import time
 from collections.abc import Iterable
 from re import Match
 from typing import Any, Callable, Optional, Union
 
 from urllib3.util import Retry
 
+try:
+    # If urllib3~=2.0 is installed
+    from urllib3 import BaseHTTPResponse
+except ImportError:
+    # If urllib3~=1.0 is installed
+    from urllib3 import HTTPResponse as BaseHTTPResponse
+
+
 from ogr.abstract import AnyComment, Comment
 
 
 class CustomRetry(Retry):
     """
     Custom Retry class that includes 403 in RETRY_AFTER_STATUS_CODES
     so that Retry-After headers are respected for 403 errors.
+
+    Also handles GitHub rate limit headers (X-RateLimit-Reset) when
+    Retry-After is not present.
     """
 
     # Include 403 in the list of status codes that respect Retry-After header
     RETRY_AFTER_STATUS_CODES = frozenset([413, 429, 503, 403])
 
+    def get_retrylimit_reset(self, response: BaseHTTPResponse) -> Optional[float]:
```

Comment:
Why is it called `retrylimit_reset` and not `ratelimit_reset`? Is that on purpose?